pkg/translator/prometheus/normalize_name_test.go

```diff
@@ -194,6 +194,7 @@ func TestBuildPromCompliantNameWithNormalize(t *testing.T) {
 	require.Equal(t, "system_io_bytes_total", BuildPromCompliantName(createCounter("system.io", "By"), ""))
 	require.Equal(t, "system_network_io_bytes_total", BuildPromCompliantName(createCounter("network.io", "By"), "system"))
 	require.Equal(t, "_3_14_digits", BuildPromCompliantName(createGauge("3.14 digits", ""), ""))
+	require.Equal(t, "envoy__rule_engine_zlib_buf_error", BuildPromCompliantName(createGauge("envoy__rule_engine_zlib_buf_error", ""), ""))
```

Comment:
These unit tests are performed with normalization feature gate ON, in which case we don't want to allow double underscores. Please change to: `require.Equal(t, "envoy_rule_engine_zlib_buf_error", ...`